gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1065,13 +1065,14 @@ public void run() {
             callbacks.getStoragePoolSize(true);
       }
       this.memorySize = memorySize;
-      if (memorySize > 0) {
+      if (memorySize > 0 && CallbackFactoryProvider.getStoreCallbacks().isEnterpriseEdition()) {
```

Comment:
Instead of silently ignoring memory-size when enterprise edition is not running, better to throw an exception inside the memorySize > 0 block:

```java
        if (!GemFireVersion.isEnterpriseEdition()) {
          throw new IllegalArgumentException(
              "The off-heap column store feature is not supported in SnappyData OSS version.");
        }
```